### DISCUSSION TITLE
Ability to hit multiple keys in Mania on touch screens

### DISCUSSION THREAD

**Discussion by smileyhead** [ppy/osu#31235] [Ideas]
I had the idea to add piano-like extra keys to the touch layout for Mania to make it possible to hit two adjacent keys with one finger. This would improve the accessibility of Mania on touch screen devices – for example, when playing in 4-key mode, it would be possible to hit 3 or 4 notes at the same time with just two thumbs.

For clarity, I am not requesting the ability to slide your fingers onto adjacent keys in streams; just the ability to press specifically two adjacent keys with the same finger.

![image](https://github.com/user-attachments/assets/767418cc-6d49-47b6-8874-2cceb0f4c44a)

**Comment by cl8n** [comment]
in a very restricted thumbs-only environment it sounds like it could be used in a cool way, but I don't think there's any avoiding that this would enable cheating, some difficult patterns can be trivialized with this setup

**Reply by smileyhead** [reply]
You can already use the same finger to hit two adjacent keys on a keyboard. I really don't think it makes a difference.

**Reply by peppy** [reply]
Dragging over a frictionless surface to hit mania "streams" may be considered a huge advantage on some beatmaps.

**Reply by smileyhead** [reply]
That sounds like a different issue. Dragging over keys is not possible, currently. I am proposing a way to hit just two adjacent keys with the same finger. No dragging.

**Reply by peppy** [reply]
That sounds very counter-intuitive..

**Reply by smileyhead** [reply]
As a Mania player, I think it's quite the opposite. In the controls' current state, you're pretty much limited to playing 2K if you're holding a phone normally, with your two thumbs touching the screen. I could set it down on a table to use my index- and middle fingers, but that would defeat the purpose of playing on a phone (might as well just get my laptop at that point).

Allowing two adjacent keys to be pressed at once would go a long way to facilitating gameplay on phones, in my opinion.

**Reply by ziv-vy** [reply]
You can just get a flat membrane keyboard and it will do the same thing. So definetly not cheating.

**Reply by ziv-vy** [reply]
This will probably be fixed via #32101. Not sure tho.

**Reply by smileyhead** [reply]
Probably not; that bug report is specifically for Standard.
